include/modem/location.h

```diff
@@ -0,0 +1,299 @@
+	LOC_EVT_ERROR,
+	/** GNSS is requesting A-GPS or P-GPS assistance data. */
+	LOC_EVT_GNSS_ASSISTANCE_REQUEST
+};
+
+/** Positioning accuracy enumerator. */
+enum loc_accuracy {
+	/** Allow lower accuracy to conserve power. */
+	LOC_ACCURACY_LOW,
+	/** Normal accuracy. */
+	LOC_ACCURACY_NORMAL,
+	/** Allow higher power consumption to improve accuracy. */
+	LOC_ACCURACY_HIGH
+};
+
+/** Positioning service. */
+enum loc_service {
+	/**
+	 * @brief Use any location service that has been configured to be available.
+	 *
+	 * @details This is especially useful when only one service is configured but can be used
+	 * even if many are available.
+	 */
+	LOC_SERVICE_ANY,
+	/** nRF Cloud location service. */
+	LOC_SERVICE_NRF_CLOUD,
+	/**Here location service. */
+	LOC_SERVICE_HERE,
+	/**Skyhook location service. */
+	LOC_SERVICE_SKYHOOK
+};
+
+/** Date and time (UTC). */
+struct loc_datetime {
+	/** True if date and time are valid, false if not. */
+	bool valid;
+	/** 4-digit representation (Gregorian calendar). */
+	uint16_t year;
+	/** 1...12 */
+	uint8_t month;
+	/** 1...31 */
+	uint8_t day;
+	/** 0...23 */
+	uint8_t hour;
+	/** 0...59 */
+	uint8_t minute;
+	/** 0...59 */
+	uint8_t second;
+	/** 0...999 */
+	uint16_t ms;
+};
+
+/** Location data. */
+struct loc_location {
+	/** Geodetic latitude (deg) in WGS-84. */
+	double latitude;
+	/** Geodetic longitude (deg) in WGS-84. */
+	double longitude;
+	/** Location accuracy in meters (1-sigma). */
+	float accuracy;
+	/** Date and time (UTC). */
+	struct loc_datetime datetime;
+};
+
+/** Location event data. */
+struct loc_event_data {
+	/** Event ID. */
+	enum loc_event_id id;
+
+	/** Used positioning method. */
+	enum loc_method method;
+
+	union {
+		/** Current location, used with event LOC_EVT_LOCATION. */
+		struct loc_location location;
+#if defined(CONFIG_LOCATION_METHOD_GNSS_AGPS_EXTERNAL)
+		/** A-GPS notification data frame used by GNSS to let the application know it
+		 *  needs new assistance data.
+		 */
+		struct nrf_modem_gnss_agps_data_frame request;
+#endif
+	};
+};
+
+/** GNSS configuration. */
+struct loc_gnss_config {
+	/**
+	 * @brief Timeout (in seconds), i.e. how long GNSS is allowed to run when trying to
+	 * acquire a fix.
+	 *
+	 * @details Note that this is not real time as experienced by the user.
+	 * Since GNSS cannot run while LTE is operating, the running time is
+	 * counted from the instant when GNSS is allowed to start. If LTE power saving mode (PSM)
+	 * is requested, GNSS waits until the modem enters PSM before starting up, thus maximizing
+	 * uninterrupted operating window and minimizing power consumption.
+	 */
+	uint16_t timeout;
+
+	/** Desired accuracy level. */
+	enum loc_accuracy accuracy;
+
+	/**
+	 * @brief If accuracy is set to LOC_ACCURACY_HIGH, instead of using the first fix, GNSS
+	 * is allowed to run for a longer time and produce the configured number fixes before the
+	 * library outputs the current location.
```

Comment:
```suggestion
	 * @brief If accuracy is set to LOC_ACCURACY_HIGH, instead of using the first fix, GNSS
	 * is allowed to run for a longer time and produce the configured number of fixes before the
	 * library outputs the current location.
```